src/main/java/io/supertokens/webserver/api/core/UsersAPI.java

```diff
@@ -106,6 +111,22 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IO
         try {
             UserPaginationContainer users = AuthRecipe.getUsers(super.main, limit, timeJoinedOrder, paginationToken,
                     recipeIdsEnumBuilder.build().toArray(RECIPE_ID[]::new));
+
+            ArrayList<String> userIds = new ArrayList<>();
+            for (int i = 0; i < users.users.length; i++) {
+                userIds.add(users.users[i].user.id);
+            }
+            HashMap<String, String> userIdMapping = UserIdMapping.getUserIdMappingForSuperTokensUserIds(super.main,
+                    userIds);
+            if (!userIdMapping.isEmpty()) {
+                for (int i = 0; i < users.users.length; i++) {
+                    String externalId = userIdMapping.get(userIds.get(i));
+                    if (externalId != null) {
+                        users.users[i].user.id = externalId;
+                    }
+                }
+            }
+
```

Comment:
need to add test that after mapping, the user ID list order is still the same.

Reply:
Another test where we are going through multiple pages for mapped user Ids, and then checking that the map is returning correctly.